R/req-promise.R

```diff
@@ -69,6 +73,12 @@ req_perform_promise <- function(req,
   check_installed(c("promises", "later"))
   check_string(path, allow_null = TRUE)
 
+  if (missing(pool)) {
+    if (!identical(later::current_loop(), later::global_loop())) {
+      cli::cli_abort("When using {.code req_perform_promise()} within {.code later::with_temp_loop()}, {.arg pool} must be provided.")
```

Comment:
I'd write this more like:

```R
cli::cli_abort(c(
  "Must supply {.arg pool} when calling {.code later::with_temp_loop()}",
  i = "Do you need {.code pool = curl::new_pool()}?"
))
```

You shouldn't need to repeat the function name because `cli_abort()` will include it by default.

Reply:
Nice, updated in [fa217a0](https://github.com/r-lib/httr2/pull/580/commits/fa217a04382685698dd1e26b23515c1e90f9166a) liking the change!